zun_kata.md

```diff
@@ -0,0 +1,148 @@
+go get -d github.com/clearcontainers/runtime
+cd $GOPATH/src/github.com/clearcontainers/runtime
+make build-kata-system && sudo -E PATH=$PATH make install-kata-system
+sudo sed -i -e 's/^# *\(enable_debug\).*=.*$/\1 = true/g' /usr/share/defaults/kata-containers/configuration.toml
+```
+
+# Build the Kata agent and create a guest OS image. 
+
+Docker should be installed and runc should be the default runtime or else the image won't build. When DevStack was setup earlier it should have installed Docker and used runc as the default runtime. This will build the Kata agent, setup a guest OS image layout, copy the agent to the image, build the image, and copy the image to the host OS.  
+
+1. Build the Kata agent and create a rootfs image
+
+```
+go get -d github.com/kata-containers/agent
+cd $GOPATH/src/github.com/kata-containers/agent && make
+go get -d github.com/kata-containers/osbuilder
+cd $GOPATH/src/github.com/kata-containers/osbuilder/rootfs-builder
+script -fec 'sudo -E GOPATH=$GOPATH USE_DOCKER=true ./rootfs.sh clearlinux'
+```
+
+2. Add the agent to the image and build it.
+
+```
+sudo install -o root -g root -m 0755 -t rootfs/bin ../../agent/kata-agent
+sudo install -o root -g root -m 0444 ../../agent/kata-agent.service rootfs/usr/lib/systemd/system/
+sudo install -o root -g root -m 0444 ../../agent/kata-containers.target rootfs/usr/lib/systemd/system/
+
+cd $GOPATH/src/github.com/kata-containers/osbuilder/image-builder
+script -fec 'sudo -E USE_DOCKER=true ./image_builder.sh ../rootfs-builder/rootfs'
+```
+
+3. Install the image
+
+```
+commit=$(git log --format=%h -1 HEAD)
+date=$(date +%Y-%m-%d-%T.%N%z)
+
+image="kata-containers-${date}-${commit}"
+sudo install -o root -g root -m 0640 -D kata-containers.img "/usr/share/kata-containers/${image}"
+(cd /usr/share/kata-containers && sudo rm -f kata-containers.img && sudo ln -s "$image" kata-containers.img)
+
+sudo ln -s /usr/share/clear-containers/vmlinux.container /usr/share/kata-containers/
+sudo ln -s /usr/share/clear-containers/vmlinuz.container /usr/share/kata-containers/
+ ```
+ 
+# Update Docker with new Kata container runtime
+
+This just replaces the Clear Containers 2.x runtime setup with DevStack with Kata containers
+
+```
+sudo sed -i 's/"cor"/"kata-runtime"/' /etc/docker/daemon.json
+sudo sed -i 's/"\/usr\/bin\/cc-oci-runtime"/"\/usr\/local\/bin\/kata-runtime"/' /etc/docker/daemon.json
+sudo systemctl daemon-reload
+sudo systemctl restart docker
+```
+
+# Test that everything works in both Docker and OpenStack Zun
+
+```
+sudo docker run -ti --runtime kata-runtime busybox sh
+zun run --name kata --runtime kata-runtime cirros ping -c 4 8.8.8.8
+zun list
+zun logs kata
+
+zun delete kata
+```
+
+# Stop DevStack and clean up system (Optional)
+
+```
+/opt/stack/devstack/unstack.sh
+/opt/stack/devstack/clean.sh
+```
+
+# Restart DevStack
+
+If you already setup Kata containers and want to restart DevStack then run the commands below.
```

Comment:
Run the following commands if you already setup Kata containers and want to restart DevStack: